api/services/gameService.js

```diff
@@ -106,62 +94,38 @@ module.exports = {
-                    return Promise.all([Promise.resolve(game), p0, p1, spectators]);
-                  }
-                  return Promise.reject({ message: "Can't populate game without two players" });
-                }
-                return Promise.reject({
-                  message: "Can't populate game, because it does not have players collection",
-                });
-              })
-              // then find points
-              .then(function findPoints(values) {
-                const [game, p0, p1, spectators] = values;
-                const p0Points = cardService.findPoints({ userId: p0.id });
-                const p1Points = cardService.findPoints({ userId: p1.id });
-                return Promise.all([
-                  Promise.resolve(game),
-                  Promise.resolve(p0),
-                  Promise.resolve(p1),
-                  Promise.resolve(spectators),
-                  p0Points,
-                  p1Points,
-                ]);
-              })
-              // then format results & resolve
-              .then(function finish(values) {
-                const [game, p0, p1, spectators, p0Points, p1Points] = values;
-                const populatedP0 = new tempUser(p0, p0Points);
-                const populatedP1 = new tempUser(p1, p1Points);
-                const result = new tempGame(game, spectators, populatedP0, populatedP1);
-                return resolve(result);
-              })
-              .catch(function failed(err) {
-                reject(err);
-              })
-          );
-        }
-        reject({ message: 'gameId is required and must be a number' });
-      } else {
-        reject({ message: 'Cannot populate Game without GameId (options had no gameId)' });
+   populateGame: async function(options) {
+    try{
+      if (!options) {
+        throw new Error({ message: 'Cannot populate Game without GameId (options had no gameId)' }); 
+      }
+      if (!Object.hasOwnProperty.call(options, 'gameId') && !typeof options.gameId === 'number') {
+        throw new Error({message: 'Cannot populate Game without GameId (options had no gameId)'});
+      }
+      // find game
+      const game = await gameService.findGame({ gameId: options.gameId });
+      
+      if (!game.players) {
+        throw new Error({message: 'Cannot populate game, because it does not have players collection'});
+      }
+      if (game.players.length < 1) {
+        throw new Error({message: 'Cannot populate game without two players'});
       }
-    });
-  }, //End populateGame()
+      //find users
+      const p0 = await userService.findUser({ userId: game.players[0].id });
+      const p1 = await userService.findUser({ userId: game.players[1].id });
+      const spectatingUsers = await fetchSpectatorUsernames(game.id);
+      //find points
+      const p0Points = await cardService.findPoints({ userId: p0.id });
+      const p1Points = await cardService.findPoints({ userId: p1.id });
```

Comment:
These are all independent of each other so we can await them as a single Promise.all() and destructure the assignments here to get the same result query in in parallel instead of in series.

One slight tweak is that the cardService calls need adjusting to grab the user ids directly from the game like we do to get the users themselves (instead of off the user objects from the above queries)